torchdata/datapipes/iter/util/randomsplitter.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import random
+from typing import Dict, TypeVar, Union
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+K = TypeVar("K")
+
+
+@functional_datapipe("random_split")
+class RandomSplitterIterDataPipe(IterDataPipe):
+    r"""
+    Randomly split samples from a source DataPipe into groups(functional name: ``random_split``).
+    Since there is no buffer, only one group of samples can be accessed at any time.
+    Note that multiple iterations of this DataPipe will yield the same split for consistency across epochs.
+
+    Args:
+        source_datapipe: Iterable DataPipe being split
+        total_length: Length of the source
+        weights: Dict of weights; the length of this list determines how many output DataPipes there will be.
+        seed: random seed used to determine the split
+        target: key of the group that will be yielded in the current iteration
+
+    Example:
+        >>> from torchdata.datapipes.iter import IterableWrapper
+        >>> dp = IterableWrapper(range(10))
+        >>> train = dp.random_split(total_length=10, weights={"train": 0.5, "valid": 0.5}, seed=0, target="train")
+        >>> list(train)
+        [1, 3, 4, 5, 9]
+        >>> valid = dp.random_split(total_length=10, weights={"train": 0.5, "valid": 0.5}, seed=0, target="valid")
```

Comment:
I prefer the case without buffer. But, could we still do
`train, val = dp.random_split()` to make sure those sub-sets are mutually exclusive?

Reply:
That makes sense. I updated the PR to do that.